src/cpp/session/modules/SessionDiagnostics.R

```diff
@@ -110,46 +110,46 @@
    # to be UTF-8 encoded; however, that encoding can be lost when forming
    # this call so bring the encoding back.
    Encoding(content) <- "UTF-8"
-   
+
    splat <- strsplit(content, "\n", fixed = TRUE)[[1]]
    starts <- grep(reStart, splat, perl = TRUE)
    ends <- grep(reEnd, splat, perl = TRUE)
-   
+
    # Get start/end pairs
    pairs <- lapply(starts, function(i) {
       following <- ends[ends > i]
       if (!length(following)) return(NULL)
       c(i, following[[1]])
    })
-   
+
    # Drop NULL
    pairs[unlist(lapply(pairs, is.null))] <- NULL
-   
+
    new <- character(length(splat))
    for (pair in pairs)
    {
       start <- pair[[1]]
       end <- pair[[2]]
-      
+
       # Ignore pairs that include 'eval = FALSE'.
       if (grepl("eval\\s*=\\s*F", splat[[start]], perl = TRUE))
          next
-      
+
       # Ignore pairs that include 'engine=', assuming they're non-R chunks.
       #
       # 'Rscript' chunks would work standalone and hence the linter would not
       # properly understand that it should discard the parse tree generated
       # from prior chunks, so we just don't lint it.
       if (grepl("engine\\s*=", splat[[start]], perl = TRUE))
          next
-      
+
       # If the chunk end lies immediately after the chunk start, bail
       if (start + 1 == end)
          next
-      
+
       new[(start + 1):(end - 1)] <- splat[(start + 1):(end - 1)]
    }
-   
+   new <- new[new != ""]
```

Comment:
This line might be the culprit in https://github.com/rstudio/rstudio/issues/10984, since we rely on the blank lines being preserved so that markers are rendered in the right location.

Reply:
Yes, that was the intent of that PR -- I just tagged for review before I got a chance to actually mention the issue number. Just removing this line on its own won't work, but that paired with a small tweak to SessionCodeTools.R fixes both the original issue and the regression